Websites/webkit.org/wp-content/themes/webkit/standards-positions.php

```diff
@@ -0,0 +1,176 @@
+#content > p#filter-container > input { width:80%; margin:0 auto; border:0; display:block; box-sizing:border-box; font-size:2rem; background-position-x:0.5rem; background-position-y:1rem; color:#000 }
+#content > p#filter-container > label { width:80%; margin:0 auto; display:block; text-align:right; font-size:1.5rem; }
+#content > p#filter-container > label > select { margin-left:.5em; width:8em }
+#content > table { table-layout:fixed; border-collapse:collapse; width:80%; margin:0 auto 50px }
+#content > table th { font-weight:bold; padding:.1em .5em; text-align:left }
+#content > table th:nth-child(1) { width:3em }
+#content > table th:nth-child(3) { width:10em }
+#content > table th:nth-child(4) { width:8em }
+#content > table th:nth-child(5) { width:10em }
+#content > table th:nth-child(6) { width:4em }
+#content > table th:nth-child(7) { width:4em }
+#content > table td { padding:.2em .5em }
+#content > table td:nth-child(4) { text-align:center }
+#content > table td:nth-child(4) > span { display:block; border-radius:15px/20px }
+#content > table td.oppose > span { background:#e40303; color:#fff }
+#content > table td.support > span { background:#008026; color:#fff }
+#content > table td.neutral >span { background:#ffed00; color:#000 }
+#content > table td.not-considering > span { background:#ff8c00; color:#000 }
+#content > table td.blocked > span { background:#784f17; color:#fff }
+#content > table > tbody > tr { background:none }
+#content > table > tbody > tr:nth-child(even) { background: #fff }
+#content > table > tbody > tr:target { background:yellow }
+
+@media only screen and (max-width: 1200px) {
+  #content > table th:nth-child(2),
+  #content > table td:nth-child(2) { width:10em }
+}
+@media only screen and (max-width: 1024px) {
+  #content > table th:nth-child(5),
+  #content > table td:nth-child(5) { display:none }
+  #content > table th:nth-child(6),
+  #content > table td:nth-child(6) { display:none }
+}
+
+@media only screen and (max-width: 800px) {
+  #content > table th:nth-child(3),
+  #content > table td:nth-child(3) { display:none }
+}
+</style>
+
+<main id=content>
+  <h1><a href="/">Standards Positions</a></h1>
+
+  <p>Enable JavaScript for an interactive summary table of WebKit's standards positions. Failing that, browse the <a href="https://github.com/WebKit/standards-positions">standards-positions GitHub repository</a> directly.</p>
+</main>
+
+<script>
+(function () { // Begin closure
+
+function createElement(name, attributes = {}, child=null) {
+  const element = document.createElement(name);
+  for (const name in attributes) {
+    element.setAttribute(name, attributes[name]);
+  }
+  if (child !== null) {
+    element.append(child);
+  }
+  return element;
+}
+
+function addElement(parent, name, attributes = {}, child=null) {
+  return parent.appendChild(createElement(name, attributes, child));
+}
+
+function createStandardsPositionsTable() {
```

Comment:
This seems to do al to of hard coded work. Why most of the static things as HTML above and then just render in the dynamic aspects. Should make things a bit easier to read/maintain.

Reply:
That would probably create a weird experience when JS is off.